### PR TITLE
Clamp 32-bit varints to 5 bytes to fix a fuzz failure.

### PR DESCRIPTION
**Pull request by haberman**

### CODE REVIEW

File: upb/decode.c [hunk 1]
```diff
@@ -221,11 +221,18 @@ static const char *decode_varint64(upb_decstate *d, const char *ptr,
 UPB_FORCEINLINE
 static const char *decode_varint32(upb_decstate *d, const char *ptr,
                                    uint32_t *val) {
-  uint64_t u64;
-  ptr = decode_varint64(d, ptr, &u64);
-  if (u64 > UINT32_MAX) decode_err(d);
-  *val = (uint32_t)u64;
-  return ptr;
+  uint64_t byte = (uint8_t)*ptr;
```
Comment (gerben-s) [inline upb/decode.c:224]: I don't think that this according to the spec. 

What you need is a decode_tag that does not read more than 5 bytes

Comment (haberman) [reply upb/decode.c:224]: This function is only used for tags and delimited field lengths.

Comment (gerben-s) [reply upb/decode.c:224]: perhaps a different name to reflect that, but that's up to you. It's definitely an error to accidentally use these for parsing fields.

Comment (haberman) [reply upb/decode.c:224]: I changed it to `decode_tag()` and switched parsing of delimited lengths to use `decode_varint64()` instead.

It makes me realize the spec is not clear about how many bytes of over-encoded varint to accept in different contexts.

**Review by gerben-s — APPROVED** [review]

**Pull request merged**